src/components/collapsible-motion/collapsible-motion.js

```diff
@@ -17,144 +33,144 @@ const createClosingAnimation = height =>
-          <ToggleAnimation isOpen={isOpen} toggle={toggle}>
-            {({
-              animation,
-              containerStyles,
-              toggle: animationToggle,
-              registerContentNode,
-            }) => (
-              <ClassNames>
-                {({ css }) => {
-                  let animationStyle = {};
-
-                  if (animation) {
-                    // By calling `css`, emotion injects the required CSS into the document head.
-                    // eslint-disable-next-line no-unused-expressions
-                    css`
-                      animation: ${animation} 200ms forwards;
-                    `;
-                    animationStyle = {
-                      animation: `${animation.name} 200ms forwards`,
-                    };
-                  }
-
-                  return this.props.children({
-                    isOpen,
-                    containerStyles: {
-                      ...containerStyles,
-                      ...animationStyle,
-                    },
-                    toggle: animationToggle,
-                    registerContentNode,
-                  });
-                }}
-              </ClassNames>
-            )}
-          </ToggleAnimation>
-        )}
-      </Collapsible>
-    );
+
+  return [animation, containerStyles, handleToggle, nodeRef];
+};
+
+const CollapsibleMotion = props => {
+  const isControlledComponent = !isNil(props.isClosed);
+
+  if (isControlledComponent) {
+    return <ControlledCollapsibleMotion {...props} />;
   }
-}
+
+  return <UncontrolledCollapsibleMotion {...props} />;
+};
+
+const ControlledCollapsibleMotion = props => {
+  const [
+    animation,
+    containerStyles,
+    animationToggle,
+    registerContentNode,
+  ] = useToggleAnimation(!props.isClosed, props.onToggle);
+
+  return (
+    <ClassNames>
+      {({ css }) => {
+        let animationStyle = {};
+
+        if (animation) {
+          // By calling `css`, emotion injects the required CSS into the document head.
+          // eslint-disable-next-line no-unused-expressions
+          css`
+            animation: ${animation} 200ms forwards;
+          `;
+          animationStyle = {
+            animation: `${animation.name} 200ms forwards`,
+          };
+        }
+
+        return props.children({
+          isOpen: !props.isClosed,
+          containerStyles: {
+            ...containerStyles,
+            ...animationStyle,
+          },
+          toggle: animationToggle,
+          registerContentNode,
+        });
+      }}
+    </ClassNames>
+  );
+};
+
+ControlledCollapsibleMotion.displayName = 'ControlledCollapsibleMotion';
+ControlledCollapsibleMotion.propTypes = collapsibleMotionPropTypes;
+
+const UncontrolledCollapsibleMotion = props => {
+  const [isOpen, toggle] = useToggleState(!props.isDefaultClosed);
+
+  const [
+    animation,
+    containerStyles,
+    animationToggle,
+    registerContentNode,
+  ] = useToggleAnimation(isOpen, toggle);
```

Comment:
Yeah, it makes it more readable, easy to understand. We could move the toggle logic into `useToggleAnimation` though

Reply:
The reason I don't put the toggle logic into useToggleAnimation is because this used for both the controlled and uncontrolled components. If we put the toggle logic in there, then the hook will be "conditional"